File: Arrays_and_Hashing/242-Valid-Anagram.py
class Solution:
    def isAnagram(s, t):
        a = {}
        b = {}
        for i in s:
            if i in a:
                a[i]+=1
            else:
                a[i]=1
        for i in t:
            if i in b:
                b[i]+=1
            else:
                b[i]=1
        if a == b:
            return True
        else:
            return False
s = 'rat'
t = 'tar'
solutionInstantiate = Solution()
print(solutionInstantiate.isAnagram(s, t))

File: Arrays_and_Hashing/242-Valid-Anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

 //1. Loop through s to get letters, but first need to split
 //2. Match s letters with t letters through hashmap
 var isAnagram = function(s, t) {
  let splitStr = s.split('');
  let splitStr2 = t.split('');

  if (splitStr.length !== splitStr2.length) {
      return false;
  };

  for (let x = 0; x<splitStr.length; x++) {
      let position = splitStr2.indexOf(splitStr[x]);
      console.log(position);
      if (position === -1) {
          return false;
      } else {
          splitStr2.splice(position, 1);
      }
  };
  return true;
};

console.log(isAnagram('anagram', 'nagaram'));
console.log(isAnagram('rat', 'car'));

